jsonschema-core.md

```diff
@@ -436,10 +434,8 @@ vocabularies MUST NOT define any keywords which begin with this prefix.
 
 #### Handling of unrecognized or unsupported keywords {#unrecognized}
 
-Implementations SHOULD treat keywords they do not recognize, or that they
-recognize but do not support, as annotations, where the value of the keyword is
-the value of the annotation. Whether an implementation collects these
-annotations or not, they MUST otherwise ignore the keywords.
+Implementations MUST refuse to process schemas which contain keywords they do
+not recognize, or that they recognize but do not support.
```

Comment:
The wording here _feels_ contradictory although I get that it isn't and why it isn't. I get that `x-` keywords are considered "recognized", but I'm not sure we're making that clear. Maybe we need to be more clear what is considered "recognized" or maybe we need to say it different way that's harder to misunderstand.

Reply:
@Relequestual had this same concern in [Slack](https://json-schema.slack.com/archives/CT7FF623C/p1717978845635409).  It's addressed in #1518.

Reply:
I'm not convinced the rewording in #1518 really addresses this concern. Not enough anyway. I think that rewording here is probably a better approach. Something like,

> Implementations MUST refuse to evaluate schemas which contain keywords they do not know how to process.

I think that's clear, covers all the bases, and avoids confusion about whether `x-` counts as a recognized keyword. Implementations know how to process `x-` keywords, so it's clear this statement isn't referring to those keywords.

Reply:
That also addresses implementations which choose to support custom keywords outside of vocabs.

Reply:
Updated with one tweak.